docs/book/how-to/snapshots/snapshots.md

```diff
@@ -0,0 +1,300 @@
+
+Once you run the snapshot, a new run will be executed on the same stack as the original run.
+
+### Using the REST API
+
+To run a snapshot through the REST API, you need to make a series of calls:
+
+1. First, get the pipeline ID:
+
+```bash
+curl -X 'GET' \
+  '<YOUR_ZENML_SERVER_URL>/api/v1/pipelines?hydrate=false&name=<PIPELINE-NAME>' \
+  -H 'accept: application/json' \
+  -H 'Authorization: Bearer <YOUR-TOKEN>'
+```
+
+2. Using the pipeline ID, get the snapshot ID:
+
+```bash
+curl -X 'GET' \
+  '<YOUR_ZENML_SERVER_URL>/api/v1/pipeline_snapshots?hydrate=false&logical_operator=and&page=1&size=20&pipeline_id=<PIPELINE-ID>' \
+  -H 'accept: application/json' \
+  -H 'Authorization: Bearer <YOUR-TOKEN>'
+```
+
+3. Finally, trigger the snapshot:
+
+```bash
+curl -X 'POST' \
+  '<YOUR_ZENML_SERVER_URL>/api/v1/pipeline_snapshots/<SNAPSHOT-ID>/runs' \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: Bearer <YOUR-TOKEN>' \
+  -d '{
+  "steps": {"model_trainer": {"parameters": {"model_type": "rf"}}}
+}'
+```
+
+{% hint style="info" %}
+Learn how to get a bearer token for the curl commands:
+
+- For Workspace/OSS API: use [temporary API tokens](https://docs.zenml.io/how-to/manage-zenml-server/connecting-to-zenml/connect-with-an-api-token) or [service accounts + API keys](https://docs.zenml.io/how-to/manage-zenml-server/connecting-to-zenml/connect-with-a-service-account).
+- For Pro management API (`cloudapi.zenml.io`): use [Pro API tokens](https://docs.zenml.io/api-reference/pro-api/getting-started#programmatic-access-with-api-tokens).
+{% endhint %}
+
+## Advanced Usage: Running Snapshots from Other Pipelines
+
+You can run snapshots from within other pipelines, enabling complex workflows. There are two ways to do this:
+
+### Method 1: Trigger by Pipeline Name (Uses Latest Snapshot)
+
+If you want to run the latest runnable snapshot for a specific pipeline:
+
+```python
+import pandas as pd
+
+from zenml import pipeline, step
+from zenml.artifacts.unmaterialized_artifact import UnmaterializedArtifact
+from zenml.artifacts.utils import load_artifact
+from zenml.client import Client
+from zenml.config.pipeline_run_configuration import PipelineRunConfiguration
+
+
+@step
+def trainer(data_artifact_id: str):
+    df = load_artifact(data_artifact_id)
+
+
+@pipeline
+def training_pipeline():
+    trainer()
+
+
+@step
+def load_data() -> pd.DataFrame:
+    # Your data loading logic here
+    return pd.DataFrame()
+
+
+@step
+def trigger_pipeline(df: UnmaterializedArtifact):
+    # By using UnmaterializedArtifact we can get the ID of the artifact
+    run_config = PipelineRunConfiguration(
+        steps={"trainer": {"parameters": {"data_artifact_id": df.id}}}
+    )
+
+    # This triggers the LATEST runnable snapshot for the "training_pipeline" pipeline
+    Client().trigger_pipeline(pipeline_name_or_id="training_pipeline", run_configuration=run_config)
+
+
+@pipeline
```

Comment:
Maybe just for the sake of the docs to save some space?

```suggestion

@step
def trainer(data_artifact_id: str):
    df = load_artifact(data_artifact_id)

@pipeline
def training_pipeline():
    trainer()

@step
def load_data() -> pd.DataFrame:
    # Your data loading logic here
    return pd.DataFrame()

@step
def trigger_pipeline(df: UnmaterializedArtifact):
    # By using UnmaterializedArtifact we can get the ID of the artifact
    run_config = PipelineRunConfiguration(
        steps={"trainer": {"parameters": {"data_artifact_id": df.id}}}
    )

    # This triggers the LATEST runnable snapshot for the "training_pipeline" pipeline
    Client().trigger_pipeline(pipeline_name_or_id="training_pipeline", run_configuration=run_config)

@pipeline
```

Reply:
That diff is not very useful :) What I did was to turn double line gaps into single line gaps.